patterns/ecs-v1/rails

```diff
@@ -1,13 +1,13 @@
 RUUID \h{32}
 # rails controller with action
-RCONTROLLER (?<controller>[^#]+)#(?<action>\w+)
+RCONTROLLER (?<[rails][controller][class]>[^#]+)#(?<[rails][controller][action]>\w+)
 
 # this will often be the only line:
-RAILS3HEAD (?m)Started %{WORD:verb} "%{URIPATHPARAM:request}" for %{IPORHOST:clientip} at (?<timestamp>%{YEAR}-%{MONTHNUM}-%{MONTHDAY} %{HOUR}:%{MINUTE}:%{SECOND} %{ISO8601_TIMEZONE})
+RAILS3HEAD (?m)Started %{WORD:[http][request][method]} "%{URIPATHPARAM:[url][original]}" for %{IPORHOST:[client][address]} at (?<timestamp>%{YEAR}-%{MONTHNUM}-%{MONTHDAY} %{HOUR}:%{MINUTE}:%{SECOND} %{ISO8601_TIMEZONE})
 # for some a strange reason, params are stripped of {} - not sure that's a good idea.
-RPROCESSING \W*Processing by %{RCONTROLLER} as (?<format>\S+)(?:\W*Parameters: {%{DATA:params}}\W*)?
-RAILS3FOOT Completed %{NUMBER:response}%{DATA} in %{NUMBER:totalms}ms %{RAILS3PROFILE}%{GREEDYDATA}
-RAILS3PROFILE (?:\(Views: %{NUMBER:viewms}ms \| ActiveRecord: %{NUMBER:activerecordms}ms|\(ActiveRecord: %{NUMBER:activerecordms}ms)?
+RPROCESSING \W*Processing by %{RCONTROLLER} as (?<[rails][request][format]>\S+)(?:\W*Parameters: {%{DATA:[rails][request][params]}}\W*)?
```

Comment:
Careful with capturing `[rails][request][params]` here. If this is a keyword field I think this is fine. If this is an object field where each param creates new keys, there's a high likelihood of ending up with an Elasticsearch mapping explosion.

Reply:
yet again, mappings weren't provided alongside the patterns.
hmm, but now that you mention we might want to provide mapping template snippets... 

from a practical perspective I am a bit hesitant since a lot of the patterns are quite outdated - not sure it's worth the additional effort (as this is a much bigger task than initially assumed already). still, worth a discussion.

Reply:
Right. If this is simply a string field that gets the usual dynamic mapping of `text` + `keyword`, no mapping explosion will happen.